src/imitation/scripts/analyze.py

```diff
@@ -74,62 +108,174 @@ def gather_tb_directories(
+        ("algo", _get_algo_name),
+        ("env_name", lambda sd: get(sd.config, "env_name")),
+        ("n_expert_demos", lambda sd: get(sd.config, "n_expert_demos")),
+        ("run_name", lambda sd: get(sd.run, "experiment.name")),
+        (
+            "expert_return_summary",
+            lambda sd: _return_summaries(sd)["expert_return_summary"],
+        ),
+        (
+            "imit_return_summary",
+            lambda sd: _return_summaries(sd)["imit_return_summary"],
+        ),
+        ("imit_expert_ratio", lambda sd: _return_summaries(sd)["imit_expert_ratio"]),
+    ]
+)
+
+
+# If `verbosity` is at least the length of this tuple, then we use all table_entry_fns
+# as columns of table.
+# Otherwise, use only the subset at index `verbosity`. The subset of columns is
+# still arranged in the same order as in the `table_entry_fns` OrderedDict.
+table_verbosity_mapping: List[Set[str]] = []
+
+# verbosity 0
+table_verbosity_mapping.append(
+    {
+        "algo",
+        "env_name",
+        "expert_return_summary",
+        "imit_return_summary",
+    }
+)
+
+# verbosity 1
+table_verbosity_mapping.append(table_verbosity_mapping[-1] | {"n_expert_demos"})
+
+# verbosity 2
+table_verbosity_mapping.append(
+    table_verbosity_mapping[-1]
+    | {"status", "imit_expert_ratio", "exp_command", "run_name"}
+)
+
+
+def _get_table_entry_fns_subset(table_verbosity: int) -> sd_to_table_entry_type:
+    assert table_verbosity >= 0
+    if table_verbosity >= len(table_entry_fns):
+        return table_entry_fns
+    else:
+        keys_subset = table_verbosity_mapping[table_verbosity]
+        result = OrderedDict()
+        for k, v in table_entry_fns.items():
+            if k not in keys_subset:
+                continue
+            result[k] = v
+        return result
+
+
 @analysis_ex.command
 def analyze_imitation(
-    source_dir: str,
-    run_name: Optional[str],
-    env_name: Optional[str],
-    skip_failed_runs: bool,
     csv_output_path: Optional[str],
-    verbose: bool,
+    tex_output_path: Optional[str],
+    print_table: bool,
+    table_verbosity: int,
 ) -> pd.DataFrame:
     """Parse Sacred logs and generate a DataFrame for imitation learning results.
 
     Args:
-      source_dir: A directory containing Sacred FileObserver subdirectories
-        associated with the `train_adversarial` Sacred script. Behavior is
-        undefined if there are Sacred subdirectories associated with other
-        scripts.
-      run_name: If provided, then only analyze results from Sacred directories
-        associated with this run name. `run_name` is compared against the
-        "experiment.name" key in `run.json`.
-      skip_failed_runs: If True, then filter out runs where the status is FAILED.
-      csv_output_path: If provided, then save a CSV output file to this path.
-      verbose: If True, then print the dataframe.
+        source_dir: A directory containing Sacred FileObserver subdirectories
```

Comment:
Argument does not exist

Reply:
Yeah, this argument isn't part of the function signature, though it is required for this Sacred command to run correctly (It's captured by another command).

I'll add something to the docstring with the effect of "you also need to look at the captured arguments for ____".

Reply:
:+1: